v3/tasks/20250321-v3-vault-factory-v2/index.ts

```diff
@@ -0,0 +1,86 @@
+import { VaultFactoryDeployment } from './input';
+import { Task, TaskMode, TaskRunOptions } from '@src';
+import { ethers } from 'hardhat';
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+export default async (task: Task, { force, from }: TaskRunOptions = {}): Promise<void> => {
+  const input = task.input() as VaultFactoryDeployment;
+
+  const vaultFactoryArgs = [
+    input.Authorizer,
+    input.pauseWindowDuration,
+    input.bufferPeriodDuration,
+    input.minTradeAmount,
+    input.minWrapAmount,
+    ethers.utils.keccak256(input.vaultCreationCode),
+    ethers.utils.keccak256(input.vaultExtensionCreationCode),
+    ethers.utils.keccak256(input.vaultAdminCreationCode),
+  ];
+
+  const vaultFactory = await task.deployAndVerify('VaultFactory', vaultFactoryArgs, from, force);
+
+  // Sanity check that the factory was deployed correctly.
+  const vaultAddress = await vaultFactory.getDeploymentAddress(input.salt);
+  if (vaultAddress !== input.targetVaultAddress) {
+    throw Error('Incorrect target address');
+  }
+
+  // Deploy the ProtocolFeeController from the artifact, as it won't be there yet on new chains.
+  // Must be AFTER the VaultFactory, or the deployer account nonce will be incorrect.
+  new Task('20250214-v3-protocol-fee-controller-v2', TaskMode.READ_ONLY);
+  const args = [vaultAddress, input.InitialGlobalProtocolSwapFee, input.InitialGlobalProtocolYieldFee];
+
+  const protocolFeeController = await task.deployAndVerify('ProtocolFeeController', args, from, force);
+
+  // Deploy the Vault contracts.
+  const deployTransaction = await task.deployFactoryContracts(
+    await vaultFactory.populateTransaction.create(
+      input.salt,
+      vaultAddress,
+      protocolFeeController.address,
+      input.vaultCreationCode,
+      input.vaultExtensionCreationCode,
+      input.vaultAdminCreationCode,
+      { gasLimit: 17e6 }
```

Comment:
```suggestion
      { gasLimit: 15e6 }
```

Reply:
We need to check this; limit should be below 15M with these changes (i.e. deploying protocol fee controller outside this tx)

Reply:
Works with 15!